OpenTrons/opentrons-api
# feat(shared-data, api): return latest pipette version from pipetteName

closes AUTH-357

# Overview

Previously, if you have access to only the `pipetteName`, the pipette definitions you get back are the first pipette version for the generation. For example, if you have a `p1000_single_flex`, you would get the pipette definitions for `3_0`. After speaking with Seth, Andy, Ed, and Sanniti, we decided that this is not great because 1) version `.0`s are prototypes and aren't in the wild and 2) the latest versions have the most up to date specs for things such as `flowRate`s. ([slack](https://opentrons.slack.com/archives/C06M6R72UDS/p1713454131088679) conversation about this if interested)

We thought about making the prototype versions the "perfect definitions" but that requires maintenance and it is much easier in the long run just to return the latest version that exists.

This change is already implemented in the frontend (see `getPipetteV2Specs`), so Protocol Designer (et al) will have the latest flow rates 🚀 .

But, we needed to align in the api as well so `opentrons-simulate` and analysis also have the latest data. This PR updates the `version_from_generation` def in `shared-data` to return the latest version. 

Note: the downfall to the current approach is there are a few api tests that will need to be updated whenever the specific pipette has a new version - see the PR for which tests those are. 

# Test Plan

upload this to the app on edge and look at the run log - the aspirate/dispense flow rate should be 160 ul/sec, which is the 1000uL tip default flow rate for the `3_0` model.

then upload this to the app on this branch and look at the run log  - the aspirate/dispense flow rate should be 716 ul/sec, which is the 1000uL tip default flow rate for the `3_6` model.

```
requirements = {
	"robotType": "Flex",
	"apiLevel": "2.16"
}

def run(ctx):
  tip_rack2 = ctx.load_labware("opentrons_flex_96_tiprack_1000ul", "D3")
  instrument = ctx.load_instrument('flex_1channel_1000', tip_racks=[tip_rack2], mount="left")
  
  my_pcr_plate = ctx.load_labware('nest_96_wellplate_200ul_flat', "C2")
  my_other_plate = ctx.load_labware('nest_96_wellplate_200ul_flat', "C1")
  ctx.load_trash_bin("A3")
    
  instrument.pick_up_tip()  
  instrument.aspirate(50, my_pcr_plate.wells_by_name()["A4"])
  instrument.dispense(20, my_other_plate.wells_by_name()["A2"])

  instrument.drop_tip()
```

# Changelog

- refactor `version_from_generation`
- update the `test_convert_pipette_name`
- update all affected tests that now return the latest pipette version from the `pipetteName`
- update the low volume default max flow rate for the `p50_single_3.6` pipette definition to match the ui max flow rate (see https://github.com/Opentrons/opentrons/pull/14859 for details on why)

# Review requests

Please review the code and logic and let me know if there is anything else I need to change?

# Risk assessment

low
